src/helper/class-ee-site.php

```diff
@@ -514,21 +516,48 @@ protected function init_le( $site_url, $site_fs_path, $wildcard = false ) {
 	 *
 	 * @param string $site_url  Name of site
 	 * @param $wildcard         Wildcard cert required?
+	 * @param bool   $allow_le  Allow LetsEncrypt
 	 *
 	 * @return array
 	 */
-	private function get_cert_domains( string $site_url, $wildcard ): array {
+	private function get_cert_domains( string $site_url, $wildcard, $allow_le = false ): array {
 
 		$domains = [ $site_url ];
 		if ( $wildcard ) {
 			$domains[] = "*.{$site_url}";
 		} else {
-			$domains[] = $this->get_www_domain( $site_url );
+
+			if ( true === $allow_le ) {
+				$domains[] = $this->get_www_domain( $site_url );
+			}
 		}
 
 		return $domains;
 	}
 
+	/**
+	 * Check www is working with site domain.
+	 *
+	 * @return bool
+	 */
+	protected function check_www_subdomain():bool {
+		$random_string = random_password();
+		file_put_contents( $this->site_data['site_fs_path'] . '/app/src/ssl_check.php', $random_string );
```

Comment:
Any specific reason to keep this as `.php` file? If a random string is only going in better make it `.html` file. Also, by keeping it `.php` file it won't work for `site-type-html`.

Reply:
Making it `.txt` for convenience.